File: src/main/java/com/musinsa/subject/model/common/Fail.java
package com.musinsa.subject.model.common;

import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.EqualsAndHashCode;
import lombok.Getter;

@Getter
@EqualsAndHashCode(callSuper = true)
public final class Fail extends Result {

    @JsonInclude(JsonInclude.Include.NON_NULL)
    private String reason;

    public Fail(String message) {
        super(-1, message);
    }

    public Fail(String message, String reason) {
        super(-1, message);
        this.reason = reason;
    }

    public Fail(int code, String message, String reason) {
        super(code, message);
        this.reason = reason;
    }

}

File: src/main/kotlin/com/musinsa/subject/model/common/Fail.kt
package com.musinsa.subject.model.common

import com.fasterxml.jackson.annotation.JsonInclude

class Fail : Result {
    @JsonInclude(JsonInclude.Include.NON_NULL)
    val reason: String?

    constructor(message: String) : super(-1, message) {
        this.reason = null
    }

    constructor(message: String, reason: String) : super(-1, message) {
        this.reason = reason
    }

    constructor(code: Int, message: String, reason: String?) : super(code, message) {
        this.reason = reason
    }
}
